### PR TITLE
review file descriptor leak

### PR DESCRIPTION
**Pull request by alejandraferreirovidal**
- https://github.com/jenkinsci/parameterized-remote-trigger-plugin/pull/32#discussion_r172029803
Causes file descriptor leak. Same as original code though...

### CODE REVIEW

**Comment by alxsap** [comment]
From my understanding chained streams and readers close each other, meaning it should be enough to close the outer one. But maybe I'm wrong 😕

File: src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteBuildConfiguration.java [hunk 1]
```diff
@@ -249,18 +251,20 @@ public void setParameterFile(String parameterFile) {
...
                 parameterList.add(sCurrentLine);
             }
         } catch (IOException e) {
             context.logger.println(String.format("[WARNING] Failed loading parameters: %s", e.getMessage()));
         } finally {
             try {
-                if (br != null) {
-                    br.close();
-                }
+                if (fileInputStream != null) fileInputStream.close();
```
Comment (alxsap) [inline src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/RemoteBuildConfiguration.java]: If one of the first `close` cause the Exception the others would be skipped.
I'd recommend to use `IOUtils.closeQuietly`.

**Comment by alejandraferreirovidal** [comment]
@alxsap I did change it again sorry.

You are right the BufferedReader closes the InputStreamReader, but the InputStreamReader does not close the FileInputStream, and that causes the file descriptor leak.

> BufferedReader.close()
510    public void More ...close() throws IOException {
511        synchronized (lock) {
512            if (in == null)
513                return;
514            in.close();
515            in = null;
516            cb = null;
517        }
518    }

You are right buffer reader closes the InputStreamReader.

> InputStreamReader.close()
198    public void More ...close() throws IOException {
199        sd.close();
200    }

The InputStreamReader closes the StreamDecoder.

The file descriptor leak is caused by not closing the FileInputStream.

> FileInputStream.close()
315    public void More ...close() throws IOException {
316        synchronized (closeLock) {
317            if (closed) {
318                return;
319            }
320            closed = true;
321        }
322        if (channel != null) {
323           channel.close();
324        }
325
326        fd.closeAll(new Closeable() {
327            public void More ...close() throws IOException {
328               close0();
329           }
330        });
331    }

**Comment by alejandraferreirovidal** [comment]
Solved with https://github.com/jenkinsci/parameterized-remote-trigger-plugin/commit/4f482c903c55124fdb6a90b61434b75e2a3d7620
hudson.FilePath.read() opens and closes the FileInputStream.

**Pull request closed without merging**